src/Security/Authentication/Negotiate/src/NegotiateHandler.cs

```diff
@@ -319,7 +319,6 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         if (OperatingSystem.IsWindows() && identity is WindowsIdentity winIdentity)
         {
             user = new WindowsPrincipal(winIdentity);
-            Response.RegisterForDispose(winIdentity);
```

Comment:
See the comments on lines 315-6. If NegotiateState returns us the same instance each time, we need to make a copy of it for use on this request.

Reply:
I understood that as the `WindowsPrincipal` object being mutable and that's still created per request. However, if the identity itself is modified too then a copy would be in order.

Reply:
Yes, the identity is mutable and needs to be copied.